Updated Player.java

Danielandersoon opened the pull request:
- Fixed issue with player movement where diagonal movement would be faster than linear movement by using a Vector2 which is normalized then multiplied by the movementSpeed attribute.
- Moved the key bindings to top of file for readability.
- Added sprint button which applies a scalar to the movement speed when SPRINT is true.
- Fixed issue with the totalTime where it would increment indefinitely causing a floating point accuracy issue.

Danielandersoon committed:
Updated Player.java:
- Fixed issue with player movement where diagonal movement would be faster than linear movement by using a Vector2 which is normalized then multiplied by the movementSpeed attribute.
- Moved the key bindings to top of file for readability.
- Addred sprint button which applies a scalar to the movement speed wehn SPRINT is true.
- Fixed issue ith the totalTime where it would increment indefinitely causing a floating poind accuracy issue.

tommyettinger commented on source code/core/src/no/sandramoen/terfenstein3D/actors/characters/Player.java:
```diff
@@ -139,26 +153,39 @@ private void resetXPosition() {
-        if (Gdx.input.isKeyPressed(Keys.W)) {
-            moveForward(-movementSpeed * dt);
+        
+        Vector2 moveVector = new Vector2(0, 0);
```
Looks good except for this -- libGDX games try really hard to avoid allocations done every frame. You could store this as a `private static final Vector2 moveVector = new Vector2();` in Player, since Vector2 is mutable, and just `moveVector.set(0, 0)` instead of allocating here.

Danielandersoon replied:
I have implemented this. Thanks.

tommyettinger replied:
Looks good to me!

Danielandersoon committed:
Update Player.java

moved move vector to top as recommended by user tommyettinger. This avoids repeated allocations.

Slideshow776 reviewed (approved):
Excellent changes : )
Thank you so much!
